pkg/storagev2/migration/migration.go

```diff
@@ -0,0 +1,111 @@
+package migration
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+	"sort"
+)
+
+type (
+	// StepFn is a function that migrates the storage to the next version
+	StepFn func(storage.Store) error
+	// StepsMap is a map of versions and their migration functions
+	StepsMap = map[uint64]StepFn
+)
+
+// Migrate migrates the storage to the latest version
+func Migrate(s storage.Store, sm StepsMap) error {
+	if err := ValidateVersions(sm); err != nil {
+		return err
+	}
+
+	currentVersion, err := GetVersion(s)
+	if err != nil {
+		return err
+	}
+
+	for nextVersion := currentVersion + 1; ; nextVersion++ {
+		if stepFn, ok := sm[nextVersion]; ok {
+			err := stepFn(s)
+			if err != nil {
+				return err
+			}
+			err = SetVersion(s, nextVersion)
+			if err != nil {
+				return err
+			}
+		} else {
+			// version doesn't exist so we are done
+			return nil
+		}
+	}
+}
+
+// ValidateVersions checks versions if they are in order n (where n min version value), n+1, n+2, n+3... (all values are increasing orders)
+func ValidateVersions(sm StepsMap) error {
+	if len(sm) == 0 {
+		return fmt.Errorf("steps map is empty")
+	}
+	versions := make([]int, len(sm))
+	i := 0
+	for key := range sm {
+		versions[i] = int(key)
+		i++
+	}
+	sort.Ints(versions)
+
+	if (versions[i-1] - versions[0]) == i-1 {
+		return nil
+	}
+	return fmt.Errorf("missing versions")
+}
+
+type StorageVersionItem struct {
+	Version uint64
+}
+
+const storageVersionItemSize = 8
+
+// ID implements the storage.Item interface.
+func (s *StorageVersionItem) ID() string {
+	return "storage_version"
+}
+
+// Namespace implements the storage.Item interface.
+func (s StorageVersionItem) Namespace() string {
+	return "migration"
+}
+
+// Marshal implements the storage.Item interface.
+func (s *StorageVersionItem) Marshal() ([]byte, error) {
+	buf := make([]byte, storageVersionItemSize)
+	binary.LittleEndian.PutUint64(buf, s.Version)
+	return buf, nil
+}
+
+// Unmarshal implements the storage.Item interface.
+func (s *StorageVersionItem) Unmarshal(bytes []byte) error {
+	if len(bytes) != storageVersionItemSize {
+		return fmt.Errorf("invalid bytes size")
```

Comment:
The error should be factored out so you can check it in your tests. By following the convention set out in other files, you get:

```golang
// errTagIDAddressItemUnmarshalInvalidSize is returned when trying
// to unmarshal buffer that is not of size tagIDAddressItemSize.
errStorageVersionItemUnmarshalInvalidSize = errors.New("unmarshal StorageVersionItem: invalid size")
```